File: convert.js
const fs = require('fs');

function styleToDict(styleStr) {
    const parts = styleStr.split(";");
    const styles = [];
    for (const p of parts) {
        if (!p.includes(":")) continue;
        let [k, ...vParts] = p.split(":");
        let v = vParts.join(":").trim().replace(/'/g, "\\'");
        k = k.trim().replace(/-([a-z])/g, (g) => g[1].toUpperCase());
        styles.push(`${k}: '${v}'`);
    }
    return "{" + styles.join(", ") + "}";
}

function processHTML(inputFile, outputFile) {
    let content = fs.readFileSync(inputFile, "utf8");

    // Replacements
    content = content.replace(/class="/g, 'className="');
    content = content.replace(/for="/g, 'htmlFor="');
    // Events
    content = content.replace(/onclick="/g, 'onClick="');
    content = content.replace(/onmouseover="/g, 'onMouseOver="');
    content = content.replace(/onmouseout="/g, 'onMouseOut="');
    
    // Self-closing tags
    const tags = ["img", "input", "br", "hr", "meta", "link", "source"];
    for (const tag of tags) {
        const regex = new RegExp(`(<${tag}\\b[^>]*?)(?<!/)>`, "gi");
        content = content.replace(regex, "$1/>");
    }

    // Styles
    content = content.replace(/style="([^"]*)"/g, (match, p1) => {
        return `style={${styleToDict(p1)}}`;
    });

    // Style tag content to avoid JSX errors with CSS syntax
    content = content.replace(/<style>([\s\S]*?)<\/style>/g, (match, p1) => {
        const css = p1.replace(/`/g, "\\`");
        return `<style dangerouslySetInnerHTML={{__html: \`${css}\`}} />`;
    });
    
    content = content.replace(/className=""/g, '');

    // Cleanup inline handlers that are now onClick="..." but should be functions or valid expressions.
    // For now, let's just make it a Next.js Client Component string!
    const jsxContent = `
    "use client";
    import React, { useEffect } from 'react';
    
    export default function Component() {
        return (
            <>
                ${content.substring(content.indexOf('<body>') + 6, content.lastIndexOf('</body>'))}
            </>
        );
    }`;
    
    fs.writeFileSync(outputFile, jsxContent, "utf8");
}

processHTML(process.argv[2], process.argv[3]);

File: convert.py
import re
import sys

def style_to_dict(style_str):
    parts = style_str.split(";")
    styles = []
    for p in parts:
        if ":" not in p: continue
        k, v = p.split(":", 1)
        k = k.strip()
        v = v.strip().replace("'", "\\'")
        k = re.sub(r"-([a-z])", lambda m: m.group(1).upper(), k)
        styles.append(f"{k}: '{v}'")
    return "{" + ", ".join(styles) + "}"

def main():
    with open(sys.argv[1], "r", encoding="utf-8") as f:
        content = f.read()

    content = content.replace('class="', 'className="')
    content = content.replace('for="', 'htmlFor="')
    
    tags = ["img", "input", "br", "hr", "meta", "link", "source"]
    for tag in tags:
        content = re.sub(r"<(%s)\b([^>]*?)(?<!/)>" % tag, r"<\1\2/>", content, flags=re.IGNORECASE)

    def style_replacer(match):
        return f"style={{{style_to_dict(match.group(1))}}}"
    
    content = re.sub(r'style="([^"]*)"', style_replacer, content)
    
    def style_tag_replacer(match):
        css = match.group(1).replace("`", "\\`")
        return f"<style dangerouslySetInnerHTML={{{{__html: `{css}`}}}} />"
    content = re.sub(r"<style>([\s\S]*?)</style>", style_tag_replacer, content)
    
    # Check for empty classNames created by replacing class=""
    content = content.replace('className=""', '')

    with open(sys.argv[2], "w", encoding="utf-8") as f:
        f.write(content)

if __name__ == "__main__":
    main()
